### PR TITLE
ref: SentryId generates UUID

### PR DESCRIPTION
**Pull request by marandaneto**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [X] Refactoring

## :scroll: Description
ref: SentryId generates UUID

## :bulb: Motivation and Context
we don't need to have a reference of the Uuid generator everywhere, and still, it's testable.

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [X] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: dart/lib/src/protocol/event.dart [hunk 1]
```diff
@@ -9,6 +9,7 @@ import '../version.dart';
 class Event {
   /// Creates an event.
   const Event({
+    this.eventId,
```
Comment (marandaneto) [inline dart/lib/src/protocol/event.dart]: @bruno-garcia @rxlabz the problem here is:

I can't do:
`this.eventId = SentryId.newId(),`
a optional default param must be a constant.

also I cannot do `final SentryId eventId = SentryId.newId();` on the field declaration.

then the class cant be immutable (const ctor) as it is inited with a non-constant value.

I don't want to make it immutable though.

ideas?

Comment (rxlabz) [reply dart/lib/src/protocol/event.dart]: I think you can remove the `const` ctor. All the fields are final + `@immutable` are enough. Dart `const` are particular.

> "const" has a meaning that's a bit more complex and subtle in Dart.

from https://news.dartlang.org/2012/06/const-static-final-oh-my.html

Comment (marandaneto) [reply dart/lib/src/protocol/event.dart]: not really, I've tried already:

the default value of an optional parameter must be constant.dartnon_constant_default_value

Comment (marandaneto) [reply dart/lib/src/protocol/event.dart]: I found this:

```
class ClassB {
  final ClassA classA;
  ClassB({this.classA = const ClassA()}); // <- creation is const
}
```

Comment (rxlabz) [reply dart/lib/src/protocol/event.dart:12]: this should do the job :
```dart
Event({
    SentryId eventId ,
    this.loggerName,
    this.serverName,
    this.release,
    this.environment,
    this.message,
    this.transaction,
    this.exception,
    this.stackTrace,
    this.level,
    this.culprit,
    this.tags,
    this.extra,
    this.fingerprint,
    this.userContext,
    this.contexts,
    this.breadcrumbs,
    this.sdk,
  }):eventId = SentryId.newId();
```

Comment (marandaneto) [reply dart/lib/src/protocol/event.dart]: if I do:

`this.eventId = const SentryId.internal(''),`

`const SentryId.internal(this._id);` it'd work but then we'd run the same problem on SentryId class

Comment (rxlabz) [reply dart/lib/src/protocol/event.dart]: this should work
```dart
Event({
    SentryId eventId,
    this.loggerName,
    this.serverName,
    this.release,
    this.environment,
    this.message,
    this.transaction,
    this.exception,
    this.stackTrace,
    this.level,
    this.culprit,
    this.tags,
    this.extra,
    this.fingerprint,
    this.userContext,
    this.contexts,
    this.breadcrumbs,
    this.sdk,
  }) : eventId = eventId ?? SentryId.newId();
```

Comment (rxlabz) [reply dart/lib/src/protocol/event.dart]: strangely the analyzer prefers this to `: this.eventId = eventId ?? SentryId.newId();` 🤷‍♂️

Comment (marandaneto) [reply dart/lib/src/protocol/event.dart]: mmmm I didn't know about the mix of syntactic sugar + named params, nice

Comment (bruno-garcia) [reply dart/lib/src/protocol/event.dart]: Would it work to drop `this`, declare the field, add a body to the ctor and add?

`this.eventId = eventId ?? SentryId();`

Comment (marandaneto) [reply dart/lib/src/protocol/event.dart]: solved doing this: https://github.com/getsentry/sentry-dart/pull/119#discussion_r508693328

thank you both

**Pull request merged**